pkg/commands/test/output.go

```diff
@@ -102,35 +103,50 @@ func (s *stdOutputManager) Flush() error {
 }
 
 type jsonCheckResult struct {
-	Filename string   `json:"filename"`
-	Warnings []string `json:"Warnings"`
-	Failures []string `json:"Failures"`
-	Successes []string `json:"Successes"`
+	Filename  string        `json:"filename"`
+	Warnings  []interface{} `json:"Warnings"`
```

Comment:
Instead of an interface would it be possible to define a new type `[]struct{string, StructuredError}`? This would be a little more explicit and prevent us from throwing `interface{}` around

Reply:
I was actually thinking to maybe just make it `[]StructuredError` and turning the original `error` into a basic `StructuredError` as well (with just the `msg` field set)

Reply:
that sounds good to me!

Reply:
With regards to your original comment, I may have misunderstood but wouldn't the struct you were recommending require a custom marshalling function as (as I understand it) Go doesn't support union types.

Reply:
Ah sorry my comment wasn't clear! In this case we would need to define a custom `json` marshaler. 

I think your suggestion of using `StructuredError` for all exceptions sounds like the best path forward. It would reduce both the complexity of our code as well as clients which may be consuming json output from this tool.

Reply:
That is tricky, because of the `--details` flag it needs to serialize either to a `string` or a `map[string]interface{}` and having some global state external to the structure affect how it json-marshals feels like it violates a lot of best practices.

One option would be to embed the value of the flag in the type itself, i.e. something like:
```go
type StructuredError struct {
    Values map[string]interface{}
    DetailedSerialization bool
}
```
but somehow that also feels wrong...

It also feels like a lot of effort to get rid of an `[]interface{}` in a non-exported struct that's used in a single-place, only for JSON marshaling.